tests/socket-shutdown.phpt

```diff
@@ -0,0 +1,32 @@
+--TEST--
+Test of Socket::shutdown() - testing params and separate shutdown of read and write channel
+--DESCRIPTION--
+Test that creating and closing sockets works as expected on all platforms (gh issue #798)
+--SKIPIF--
+<?php
+    if(!method_exists(\Socket::class, 'shutdown')) {
+        die('skip.. Socket::shutdown() doesn\'t exist');
+    }
+--FILE--
+<?php
+    $socket = new \Socket(\Socket::AF_INET, \Socket::SOCK_STREAM, 0);
+    $socket->shutdown();
+
+    var_dump($socket->shutdown(4)); // invalid - false
```

Comment:
On Windows, this causes an `ENOTCONN` error to be emitted, which causes a `RuntimeException` to be thrown. I assume that in Winsock2, socket state is checked prior to checking the argument value [citation needed].

Reply:
@dktapps I slightly modified the test from bind()/listen() to connect(). Can you check the test on a windows machine?